app/models/liabilitycalculation/viewmodels/TaxDueSummaryViewModel.scala

```diff
@@ -125,10 +125,29 @@ object TaxDueSummaryViewModel {
         totalAnnuityPaymentsTaxCharged = calc.taxCalculation.flatMap(tc => tc.totalAnnuityPaymentsTaxCharged),
         totalRoyaltyPaymentsTaxCharged = calc.taxCalculation.flatMap(tc => tc.totalRoyaltyPaymentsTaxCharged),
         giftAidTaxChargeWhereBasicRateDiffers = calc.taxCalculation.flatMap(tc => tc.incomeTax.giftAidTaxChargeWhereBasicRateDiffers),
-        totalTaxableTransitionProfit = calc.transitionProfit.flatMap(_.totalTaxableTransitionProfit),
-        incomeTaxChargedOnTransitionProfits = calc.taxCalculation.flatMap(_.incomeTax.incomeTaxChargedOnTransitionProfits)
+        transitionProfitRow = TransitionProfitRow(
+          calc.taxCalculation.flatMap(_.incomeTax.incomeTaxChargedOnTransitionProfits),
+          calc.transitionProfit.flatMap(_.totalTaxableTransitionProfit))
       )
       case None => TaxDueSummaryViewModel()
     }
   }
 }
+
+object TransitionProfitRow {
+  def apply(incomeTaxCharged: Option[BigDecimal], totalTaxableProfit: Option[BigDecimal]): Option[TransitionProfitRow] = {
+    (incomeTaxCharged, totalTaxableProfit) match {
+      case (Some(tax), Some(profit)) => Some(TransitionProfitRow(incomeTaxCharged = tax, totalTaxableProfit = profit))
+      case (None, _) =>
+        Logger("application").warn(s"[TaxDueSummaryViewModel][TransitionProfitRow] missing incomeTaxChargedOnTransitionProfits")
+        None
+      case (_, None) =>
+        Logger("application").warn(s"[TaxDueSummaryViewModel][TransitionProfitRow] missing totalTaxableTransitionProfit")
+        None
+      case _ => None
```

Comment:
This case is not shown as being covered by the tests. I think the None, None case is matching earlier and will be logging inappropriately.

Reply:
Made the change.

Reply:
Cool